src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2832,8 +2835,7 @@ private void GenerateModuleInitializer(PEModuleBuilder moduleBeingBuilt, Diagnos
             {
                 var ilBuilder = new ILBuilder(moduleBeingBuilt, new LocalSlotManager(slotAllocator: null), OptimizationLevel.Release, areLocalsZeroed: false);
 
-                // PROTOTYPE(module-initializers): require deterministic order
-                foreach (var method in _moduleInitializerMethods)
+                foreach (MethodSymbol method in _moduleInitializerMethods.OrderBy<MethodSymbol>(LexicalOrderSymbolComparer.Instance))
```

Comment:
The documentation for `LexicalOrderSymbolComparer` indicates it's only supported for `Symbol` instances that are declared in the same container:

> (explicitly or explicitly declared in source  within the same container) 

Module initializers can be declared across containers though. Have you verified that `LexicalOrderSymbolComparer` is supported for this scenario? If so can you clean up the documentation to reflect the correctly supported behaviors?

Reply:
Yes, reading the source indicates this and `ModuleInitializersTests.MultipleInitializers_MultipleFiles` seems to demonstrate this. I will fix up the documentation.